str.c

```diff
@@ -723,6 +728,98 @@ str_replace_unprintable(struct mystr* p_str, char new_char)
   }
 }
 
+void
+str_replace_unprintable_with_hex(struct mystr* p_str)
+{
+  unsigned int ups_size = sizeof(unsigned int) * (p_str->len);
+  if (ups_size < p_str->len)
+  {
+    bug("string is to long");
+  }
+  unsigned int* ups = vsf_sysutil_malloc(ups_size);
+  unsigned int up_count = 0;
+  for (unsigned int i=0; i < p_str->len; i++)
+  {
+    if (!vsf_sysutil_isprint(p_str->p_buf[i]))
+    {
+      ups[up_count++] = i;
+    }
+  }
+  str_replace_positions_with_hex(p_str, ups, up_count);
+  vsf_sysutil_free(ups);
+}
+
+void str_replace_unprintable_with_hex_wc(struct mystr* p_str)
+{
+  unsigned int ups_size = sizeof(unsigned int) * (p_str->len);
+  if (ups_size < p_str->len)
+  {
+    bug("string is to long");
```

Comment:
Logging this is not really useful. Could it log the string with unprintable replaced with ? as before, if unable to process the original string?

Reply:
fixed